app/src/main/kotlin/org/open/smsforwarder/processing/processor/ForwardingProcessor.kt

```diff
@@ -19,7 +19,7 @@ class ForwardingProcessor @Inject constructor(
     private val authRepository: AuthRepository
 ) {
 
-    suspend fun process(messages: Array<String>) {
+    suspend fun process(messages: Array<String>, historyEntityId: Long? = null) {
```

Comment:
it;s a code smell. You don't need to change the processor logic.
You just need to pass it the message which you can get before from history entity by id. 
The rest or the process will remain untouched